src/app/song/song-action.sevice.ts

```diff
@@ -0,0 +1,29 @@
+import { Subject } from 'rxjs';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Song, Genre } from './song.model';
+import { NotificationPopupService } from '../notification/notification-popup.service'
+import { NotificationStatus, Notification } from '../notification/notification.model'
+
+
+@Injectable({providedIn: 'root'})
+export class SongActionService{
+
+    private song: Song;
+    private songLiked = new Subject<{song:Song, newNumOfLikes:number}>();
+
+    constructor(private Http: HttpClient,
+                private notificationService:NotificationPopupService){}
+
+    getSongUpdateListener() { return this.songLiked.asObservable(); }
+
+    //likesAction is 1 if liked, -1 if disliked
+    updateSongLikeStatus(song: Song, likeAction: number) {
```

Comment:
Maybe call an api called /like or /unlike and not just update the number of likes. cause then you can just override that.